### ISSUE TITLE
LaTeX request

### ISSUE THREAD

**Issue by FISHunderscore** [ForkAwesome/Fork-Awesome#54]
Is there any possibility of a LaTeX package for Fork-Awesome similar to the [FontAwesome package](https://www.ctan.org/pkg/fontawesome)?

I may attempt this myself if there is not anyone more familiar with LaTeX here.

**Comment by xuv** [comment]
I guess we could. It's a nice idea. I have no idea how to do it though. If you have that kind of knowledge, pull requests are welcome.

I see that it's pointing to a different repository than the official one from FontAwesome: https://github.com/xdanaux/fontawesome-latex

It seems we would also probably need to export the .otf version of the font for it to work.

**Comment by Atyansh** [comment]
Would definitely love to see this. I can help out though I don't know what steps are needed to make this happen.

**Comment by tessus** [comment]
Unfortunately @xdanaux seems to have abandoned the project.

But I believe it's rather easy to adjust the [generate_tex_bindings.py](https://github.com/xdanaux/fontawesome-latex/blob/master/generate_tex_bindings.py) to make it work with ForkAwesome.

However, I'd like to know what the [pdftex mapping](https://github.com/xdanaux/fontawesome-latex/blob/1a6ce6ef7bbac3656095cedd86f0b99140e201d3/generate_tex_bindings.py#L22-L470) is for. You most likely will have to adjust the array and add the additional icons.
Since I don't know why the mapping exists in the first place, I'm reluctant to look at this any further.

Other than that it's just running the script and it will create all the necessary files for LaTeX.

**Comment by tessus** [comment]
@Atyansh the first step would be to contact @xdanaux or the Latex community to figure out why there is a mapping for pdftex. As soon as you have an answer to that, we can go from there.

**Comment by tessus** [comment]
@Atyansh it's also important to know how to do the mapping. I can't figure out where the keys are coming from.
e.g.:

`"_620" : "percent", #f295`

The percent symbol/icon is unicode point f295, but what has `_620` to do with that?

**Comment by zauguin** [comment]
As an alternative to modifying fontawesome-latex, [zauguin/forkawesome](https://github.com/zauguin/forkawesome.git) is a first draft for a new forkawesome package avoiding fontspec. That avoids some problems fontspec caused for FontAwesome.

**Comment by tessus** [comment]
@zauguin awesome, this looks great. I will have to do some research into how to create a latex package and distribute it.
I've also received a few answers in `comp.text.tex`, but I haven't had any time to look into them yet.

> That avoids some problems fontspec caused

See, I don't even know what fontspec does and what issues it caused.

I really appreciate any help with this.

**Comment by zauguin** [comment]
@tessus Fontspec provides a nice user interface for loading arbitrary fonts with XeLaTeX and LuaLaTeX. It's mostly focused on text fonts, so most of it's features are not relevant for icon fonts. This caused for example xdanaux/fontawesome-latex#5, where fontspec is confused because fontawesome does not contain any lowercase characters. Similarly LaTeX complains when icons are used in bold/italic/small-caps/... text because LaTeX can not find a bold/italic/... version of fontawesome.
These are not huge problems and probably could be fixed, but there is no real advantage in using fontspec here, so why bother?

In general font packages like this are split into two parts: When the package is loaded in XeLaTeX/LuaLaTeX, they can directly read ony `otf` or `ttf` file.

In pdfTeX this is more difficult. In `comp.text.tex` were some notes already about pdfTeX supporting `ttf` fonts, but the `forkawesome-webfont.ttf` does not work here. Mostly the problem is about the `OS/2` table in the font being in version `4`, while pdfTeX does only support `OS/2` tables upto version `3`. So this does not really apply here and instead you can convert the font to PostScript Type1 (e.g. using fontforge).
In any case, pdfTeX requires an encoding and a metric file (`.tfm`). Actually you need three of each, because pdfTeX can only work with fonts with upto 256 glyphs. Of course, then some mappping on the TeX layer is required to map the icon names to the right font and the right encoding slot in the font.

In the linked repository you can see one way of generating all these files in the `gen` file.
It is not that readable because it relies a lot on Linux based shell trickery but it might give you an idea about how things work. It is adapted from my `fontawesome5` package where it works quite reliable.

If there are any further questions I'm happy to help.

**Comment by tessus** [comment]
@zauguin Thanks again for the detailed answer. This is awesome. Unfortunately I'm currently a bit busy at work and don't have much time looking into this. But you gave me a lot of good pointers and the `gen` script definitely helps. If I have any specific questions (when I will finally have time to get to it), I will bug you again. ;-)

**Comment by zauguin** [comment]
@tessus There were some problems with the old `fontawesome` package recently so I came back to this.
I think LaTeX would really benefit from a supported package which is mostly compatible with `fontawesome` (4).
So if you do not have enough time, would you agree if I upload a forkawesome package instead?

**Comment by tessus** [comment]
@zauguin I totally agree with you, but I would like to add this to our release pipeline. Do you think you could add the process to our repo, so that whenever there's a new release, the package is automatically created and uploaded?

**Comment by tessus** [comment]
@zauguin just wanted to follow up on your last comment...
